merge/wipe_fields_test.go

```diff
@@ -0,0 +1,187 @@
+			},
+		},
+		"does not own modified fields removed by prepare": {
+			liveManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(), "v1", false),
+			},
+			newManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a")), "v1", false),
+			},
+			manager:        "test",
+			liveObject:     `{"a": 1}`,
+			newObject:      `{"a": 2}`,
+			preparedObject: `{"a": 1}`,
+			expectedManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(), "v1", false),
+			},
+		},
+		"does not wipe added fields not removed by prepare": {
+			liveManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a")), "v1", false),
+			},
+			newManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a"), _P("b")), "v1", false),
+			},
+			manager:        "test",
+			liveObject:     `{"a": 1}`,
+			newObject:      `{"a": 1, "b": 2}`,
+			preparedObject: `{"a": 1, "b": 2}`,
+			expectedManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a"), _P("b")), "v1", false),
+			},
+		},
+		"does not wipe modified fields not removed by prepare": {
+			liveManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a")), "v1", false),
+			},
+			newManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a")), "v1", false),
+			},
+			manager:        "test",
+			liveObject:     `{"a": 1}`,
+			newObject:      `{"a": 2}`,
+			preparedObject: `{"a": 2}`,
+			expectedManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a")), "v1", false),
+			},
+		},
+		"does remove removed fields not reset by prepare": {
+			liveManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a"), _P("b")), "v1", false),
+			},
+			newManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a")), "v1", false),
+			},
+			manager:        "test",
+			liveObject:     `{"a": 1, "b": 2}`,
+			newObject:      `{"a": 1}`,
+			preparedObject: `{"a": 1}`,
+			expectedManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a")), "v1", false),
+			},
+		},
+		"does not remove removed fields reset by prepare": {
+			liveManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a"), _P("b")), "v1", false),
+			},
+			newManagedFields: fieldpath.ManagedFields{
+				"test": fieldpath.NewVersionedSet(_NS(_P("a")), "v1", false),
+			},
+			manager:        "test",
+			liveObject:     `{"a": 1, "b": 2}`,
+			newObject:      `{"a": 1}`,
+			preparedObject: `{"a": 1, "b": 2}`,
+			expectedManagedFields: fieldpath.ManagedFields{
+				// TODO: should `b` still be owned despite the applier not intending to own it?
+				"test": fieldpath.NewVersionedSet(_NS(_P("a") /* _P("b") */), "v1", false),
```

Comment:
should `b` still be owned despite the applier not intending to own it?

Reply:
`b` should still be owned by whoever owned it before, which is going to be a challenge :-(

Reply:
That's gonna be a problem.
Might mean it won't work this way and we'll have to do it directly in the update.
Which then means we come in a kind of chicken egg problem, because this mechanism needs the fields after update so we need to run update twice or something...